### PR TITLE
Fix preview for avro topic with schemaId aware serialization

### PR DESCRIPTION
**Pull request by dswiecki**
I've tried to write some integration test in ZookeeperMessagePreviewRepositoryTest but failed on missing dependencies to MessagePreviewLog which is inside hermes-frontend, do you see how to test it as a whole?

### CODE REVIEW

File: hermes-frontend/src/test/groovy/pl/allegro/tech/hermes/frontend/publishing/preview/MessagePreviewLogTest.groovy [hunk 1]
```diff
@@ -89,4 +109,10 @@ class MessagePreviewLogTest extends Specification {
         executorService.shutdown()
     }
 
+    static ObjectMapper createObjectMapper() {
```
Comment (pitagoras3) [inline hermes-frontend/src/test/groovy/pl/allegro/tech/hermes/frontend/publishing/preview/MessagePreviewLogTest.groovy:112]: There are no usages of this objectMapper, can you remove it?

**Review by pitagoras3 — COMMENTED** [review]
Hi @dswiecki, thank you for your contribution to Hermes repository!

File: hermes-frontend/src/test/groovy/pl/allegro/tech/hermes/frontend/publishing/preview/MessagePreviewLogTest.groovy (hunk 1 above)
Comment (dswiecki) [reply hermes-frontend/src/test/groovy/pl/allegro/tech/hermes/frontend/publishing/preview/MessagePreviewLogTest.groovy]: ok, removed

**Review by pitagoras3 — APPROVED** [review]

**Comment by dswiecki** [comment]
@pitagoras3 can you approve workflow so the pipeline can run?

**Comment by piotrrzysko** [comment]
> do you see how to test it as a whole?

It is a little bit tricky, but it is possible :). There is a module called `integration` where we put integration tests. 

The test may look like this:

```java
@Test
public void shouldReturnAvroMessageWithSchemaAwareSerialization() {
    // given
    AvroUser avroUser = new AvroUser("Bob", 50, "blue");
    Topic topic = randomTopic("avro", "returnSchemaAwareSerialization")
            .withSchemaIdAwareSerialization()
            .withContentType(AVRO)
            .build();
    TopicWithSchema topicWithSchema = topicWithSchema(topic, avroUser.getSchemaAsString());
    operations.buildTopicWithSchema(topicWithSchema);
    publisher.publish(topic.getQualifiedName(), avroUser.asBytes());
    wait.awaitAtMost(Duration.TEN_SECONDS).until(() -> management.topic().getPreview(topic.getQualifiedName()).size() == 1);

    // when
    List<MessageTextPreview> previews = management.topic().getPreview(topic.getQualifiedName());

    // then
    assertThat(previews).hasSize(1);
    assertThatJson(previews.get(0).getContent())
            .when(Option.IGNORING_EXTRA_FIELDS)
            .isEqualTo(avroUser.asJson());
}
```

Probably we can put it in `pl.allegro.tech.hermes.integration.management.TopicManagementTest`. 

Then we need to add a definition of the `preview` endpoint in `pl.allegro.tech.hermes.api.endpoints.TopicEndpoint` (for some reason it is not done yet): 

```java
@GET
@Produces(APPLICATION_JSON)
@Path("/{topicName}/preview")
List<MessageTextPreview> getPreview(@PathParam("topicName") String qualifiedTopicName);
```

We have to make `pl.allegro.tech.hermes.api.MessageTextPreview` deserializable: 

```java 
@JsonCreator
public MessageTextPreview(@JsonProperty("content") String content, @JsonProperty("truncated") boolean truncated) 
```

and enable the preview feature in the constructor of `pl.allegro.tech.hermes.integration.env.FrontendStarter`:
```java
configFactory.overrideProperty(FRONTEND_MESSAGE_PREVIEW_ENABLED, true);
configFactory.overrideProperty(FRONTEND_MESSAGE_PREVIEW_LOG_PERSIST_PERIOD, 1);
```

**Review by piotrrzysko — APPROVED** [review]

**Pull request merged**